---
title: Population structure and phenotypic variation of *Sclerotinia sclerotiorum* from dry bean in the United States
preprint: false
author:
  - name: Zhian N. Kamvar
    email: zkamvar@unl.edu
    affiliation: 1
  - name: Bimal Sajeewa Amadarasa
    email: bamadarasa2@unl.edu
    affiliation: 1
  - name: Rachana Jhala
    email: rjhala2@unl.edu
    affiliation: 1
  - name: Serena McCoy
    email: serena.mccoy@icloud.com
    affiliation: 1
  - name: James Steadman
    email: jsteadman1@unl.edu
    affiliation: 1
  - name: Sydney E. Everhart
    email: everhart@unl.edu
    affiliation: 1
    corresponding: true
affiliation:
  - code: 1
    address: Department of Plant Pathology, University of Nebraska-Lincoln, Lincoln 68583
abstract: >
  `r paste(readLines("abstract.md"), collapse = "\n  ")`
header-includes: >
  \usepackage{indentfirst}
bibliography: ssc_bibliography.bib
csl: apa.csl
output: 
  rticles::peerj_article:
    citation_package: none
    base_format: bookdown::pdf_document2 # for using \@ref()
---

```{r setup, include = FALSE, echo = FALSE}
knitr::opts_knit$set(eval.after = c('fig.cap', 'fig.scap'))
knitr::opts_chunk$set(message = FALSE, warning = FALSE)
```

```{r initiate, echo = FALSE, message = FALSE, warning = FALSE}
library("poppr")
library("tidyverse")
load("../../data/sclerotinia_16_loci.rda")
load("../../data/mcg-mlg-graph.rda")
load("../../data/heterothallic-pairings.rda")
caplist1 <- read_rds("../../data/caplist1.rds")
caplist2 <- read_rds("../../data/caplist2.rds")
dat11cc  <- clonecorrect(dat11, ~Region/Source/Host/Year)
pop_size <- table(pop(dat11))

# Utility Functions -------------------------------------------------------

# convert fractions to percents with 3 sig figs
prettify <- function(x, i) signif((x[i])*100, 3)

# Get proportion of variance per discriminant axis
paxis <- function(d, i = 1){
  res <- cumsum(d$eig)/sum(d$eig)
  prettify(1 - res, i)
}

# Get proportion of assignment probability for a given population
passp <- function(d, i = 1){
  res <- apply(d$posterior, 2, mean, na.rm = TRUE)
  prettify(res, i)
}

# Tabulate variance partitions
vartab <- function(x){
  tot <- x$tot.chi
  con <- x$constr.chi
  uco <- x$unconst.chi
  data.frame(list(Inertia = c(tot, con, uco),
                  Proportion = c(tot, con, uco)/tot))
}

# Take a character vector and create a list in english (WITH OXFORD COMMA): 
# e.g. "Juice, Eggs, and Toast"
andlist <- function(x) paste( paste(x[-length(x)], collapse = ", "), x[length(x)], sep = ", and " )

# Locus Table -------------------------------------------------------------

loc_columns <- cols(
  Locus = col_character(),
  Range = col_character(),
  Repeats = col_double(),
  allele = col_integer(),
  Hexp = col_double(),
  Evenness = col_double()
)
loc_table <- readr::read_csv("../../results/tables/locus-table.csv", 
                             col_types = loc_columns,
                             col_names = TRUE)
Amean <- mean(loc_table$allele)
Hmean <- mean(loc_table$Hexp)
Emean <- mean(loc_table$Evenness)

# DAPC data ---------------------------------------------------------------

dapc_region      <- dapc(setPop(dat11cc, ~Region), n.da = 13, n.pca = 20)
dapc_region_year <- dapc(setPop(dat11cc, ~Region/Year), n.da = 15, n.pca = 15)
dapc_host        <- dat11cc %>% 
  setPop(~Source) %>%
  popsub("wmn") %>%
  setPop(~Host) %>% 
  popsub(c("Beryl", "Bunsi", "G122")) %>%
  dapc(n.da = 2, n.pca = 20)

# RDA results -------------------------------------------------------------

RDA1R2    <- prettify(caplist1$R2$adj.r.squared)/100
RDA2R2    <- prettify(caplist2$R2$adj.r.squared)/100
anova1res <- setNames(caplist1$anova$`Pr(>F)`, rownames(caplist1$anova)) %>% na.omit()
anova2res <- setNames(caplist2$anova$`Pr(>F)`, rownames(caplist2$anova)) %>% na.omit()
RDAVAR1   <- andlist(names(anova1res))
RDAVAR2   <- andlist(gsub("Severity", "Aggressiveness", names(anova2res)))

# MLG/MCG comparison

big_haplotype <- in_reach %>% filter(N == max(N))
```


# Introduction {-}

<!-- 
ZNK: The introduction needs a lot of work. Here's an outline of what I think
     needs to go in here.
Outline:

 - *S. sclerotiorum* basic biology
 - What have genetic studies and phenotypic studies found
 - What are we missing? A detailed analysis of *S. sclerotiorum* from diverse
   regions that may potentially identify haplotypes or MCGs that may be good
   tools for testing breeding efforts.
 - Our hypotheses and specific questions.
 -->

*Sclerotinia sclerotiorum* (Lib.) de Bary is an ascomycete plant pathogen with a worldwide distribution.
This is a necrotrophic pathogen that is primarily homothallic (self-fertilization) and has the ability to survive fore more than five years in soil using melanized survival structures called sclerotia [@bolton2006sclerotinia; @sexton2006population]. 
It causes disease on more than 400 plant species belonging to 75 families [@boland1994index] including crops of major economic importance such as sunflower (*Helianthus spp.*), soybean (*Glycine max* L.), canola (*Brassica napa* L., *Brassica campestris* L.), and dry bean (*Phaseolus vulgaris* L.) [@bolton2006sclerotinia]. 
On dry bean, *S. sclerotiorum* is the causal agent of white mold, a devastating disease that can be a yield-limiting in temperate climates [@steadman1983white].

Currently, there are no known resistant cultivars of dry bean commercially available [@otto-hanson2011variation].
Organized breeding efforts have used a common-garden approach with white mold screening nurseries in dry bean production areas across the United States with additional sites in Mexico, France, and Australia [@steadman2003identification; @steadman2004identification; @steadman2005identification; @steadman2006identification; @otto2007identification; @otto2008identification; @mccoy2009use]. 
These white mold screening nurseries use no chemical or cultural treatments against *S. sclerotiorum* and employ standardized protocols for screening new cultivars for resistance to white mold. 
It has previously been shown that aggressiveness (the severity of disease symptoms on the host) is significantly different across white mold screening nursery sites in separate geographic regions [@otto-hanson2011variation].
The genetic structure and mode of reproduction in these populations, however, is currently unknown.

Understanding genetic relationships and reproduction behavior of *S. sclerotiorum* populations is beneficial for breeders seeking to develop new resistant cultivars for worldwide deployment [@mcdonald2002pathogen; @milgroom1996recombination].
In particular, genetically diverse populations with high rates of sexual reproduction are more likely to overcome host resistance. 
Most populations of *S. sclerotiorum* are predominantly clonal with low genetic diversity and have a large degree of population fragmentation [@attanayake2012genetic; @carbone2001multilocus; @cubeta1997clonality; @ekins2011population; @kohli1998random; @kohli1995clonal]. 
Some studies, however have found populations that show signatures of sexual reproduction [@atallah2004high; @attanayake2013sclerotinia; @sexton2004microsatellite; @aldrich-wolfe2015genetic].

Nearly all population genetic studies of *S. sclerotiorum* employ a macroscopic assay to determine mycelial compatibility, the ability for fungal hyphae from different colonies to appear to grow together without forming a barrier of dead cells between them (known as a barrage line) [@leslie1993fungal; @sirjusingh2001characterisation].
Mycelial compatibility has been used as a proxy for vegetative compatibility, a fungal trait controlled by several independent genes controlling the ability for two hyphae to fuse and grow as a single unit [@leslie1993fungal; @schafer2006optimized].
Because of the genetic connection to vegetative compatibility, two isolates that are mycelially compatible were considered clones [@leslie1993fungal]; but correlation with genetic markers, such as microsatellites, have shown mixed results [@attanayake2012genetic; @lehner2017independently].

In this study, we analyze and characterize the genetic and phenotypic diversity of 366 *S. sclerotiorum* isolates collected over 10 years between 2003 and 2012 from dry bean cultivars among different geographic locations in the United States, France, Mexico, and Australia. 
We wanted to know if the *S. sclerotiorum* populations from white mold screening nurseries were representative of the fields within the same region. 
As these nurseries were not treated with any chemical or cultural control of white mold, we hypothesized that these nurseries would represent the natural population of *S. sclerotiorum*.
Furthermore, we wanted to investigate the potential effect of cultivar on genetic diversity by assessing three dry bean cultivars with different levels of resistance, Beryl (great northern bean, susceptible), Bunsi (navy bean, low susceptibility), and G122 (cranberry bean, partial resistance) [@otto-hanson2011variation].
We additionally wanted to determine categorical or phenotypic variables that best predicted genetic structure and if there was correlation between multilocus haplotype and mycelial compatibility group.
Knowing what variables predict genetic structure can help direct breeding efforts.
By investigating these aims, we hope to effectively describe the population structure of *S. sclerotiorum* in the USA and make available our database of isolates for use in future dry bean breeding efforts.


# Materials and Methods {-}

## Isolate collection {-}

Several of the isolates used for this study were collected as reported in previous studies using the same methods [@otto-hanson2011variation].
Broadly, isolates were collected from two sources: white mold screening nurseries (wmn) or producer fields.
<!-- SEE: Insect or weed control? -->
White mold screening nurseries were 5m x 10m in size and maintained without application of fungicides to observe natural incidence of white mold.
The nursery plots were incorporated with a basal dressing of N:P:K = 1:3:2 and side dressing of 0:3:2 during the growing season [@steadman2003identification; @steadman2004identification; @steadman2005identification; @steadman2006identification; @otto2007identification; @otto2008identification; @mccoy2009use].

Sampling was carried out by collecting sclerotia from diseased tissue in zig-zag transects across field plots. 
Because sampling depended on disease incidence, the number of samples isolated varied from year to year.
Although the nursery locations were the same over sampling years, sampling plots within a location varied for sampling years.

Sclerotia of *S. sclerotiorum* were collected over several years from grower fields and/or wmn in 11 states of the United States of America, Mexico, France, and Australia (Table \@ref(tab:isolate-table)). 
After collection, sclerotia were stored in Petri plates lined with filter paper, then stored at $20~^{\circ}$F or -$4~^{\circ}$C. 
Sclerotia were surface-sterilized with 50% sodium hypochlorite (at least 6% NaOCL, The Clorox Company, Oakland, CA) solution for 3 min, and double rinsed with ddH$_2$O 3 min.
The sterilized sclerotia were then plated on water agar plates (16g of Bacto agar per liter of ddH$_2$O, BD Diagnostic Systems, Sparks, MD), with four to five sclerotia of each isolate separated on each plate and stored on the counter top at room temperature for 5 to 6 days. 
An 8-mm plug from a 5- or 6-day-old culture was transferred from the advancing margin of the mycelia onto a plate of Difco potato dextrose agar (PDA at 39 g/liter of ddH$_2$O) [@otto-hanson2011variation].
In combination with the 156 isolates described previously, we collected 210 isolates for a total of 366 isolates [@otto-hanson2011variation].

## Mycelial Compatibility {-}

MCG was determined as described previously through co-culturing pairs of 2-day-old isolates 2.5 cm apart on Diana Sermons (DS) Medium [@cubeta2001mycelial]. 
Incompatibility of different MCGs resulted in formation of a mycelial barrier of dead tissue (termed a barrage line) accompanied by formation of sclerotia on either side of the barrage line, indicating the limits of the isolates' growth [@otto-hanson2011variation; @kohn1990mycelial; @leslie1993fungal].
Isolates were compared in a pairwise manner for each site and then representatives among sites were compared to determine mycelial compatibility groups by scoring compatible and incompatible interactions [@otto-hanson2011variation]. 

<!-- ![
Examples of mycelial compatibility results in *S. sclerotiorum*. 
A) Compatible reaction showing aerial hyphae.
B) Incompatible reaction showing barrage reaction of dead cells at point of contact.
](../../results/figures/publication/incompatibility2.JPG){#incompatibility width=25%} -->

## Aggressiveness {-}

Aggressiveness of each isolate was assessed using a straw test as described in @otto-hanson2011variation that rated necrotic lesion size [@petzoldt1996straw; @teran2006modified].
Briefly, the straw test uses 21-day-old G122 plants as the host in a greenhouse setting. 
Clear drinking straws cut to 2.5 cm and sealed were used to place two mycelial plugs of inoculum on the host plant after removing all plant growth &gt;2.5 cm above the fourth node.
Measurements were taken 8 days later using the Modified Petzoldt and Dickson scale of 1--9, where 1 is no disease and 9 is plant death [@petzoldt1996straw; @teran2006modified].

## Microsatellite genotyping {-}

Prior to DNA extraction, isolates were grown on PDA and plugs were subsequently transferred to Potato Dextrose Broth (PDB) where they were grown until there was significant mycelial growth, but before the mycelial mat became solidified (4--5 days).
Mycelia for each plate was collected in a buchner funnel, agar plugs removed, lyophilized and pulverized manually in Whirl-pak® HDPE sampling bags (Sigma-Aldrich, St. Louis, MO). 
Lyophilized mycelia was then stored in microcentrifuge tubes at -$20~^{\circ}$C until needed for DNA extraction.
DNA from 25mg of pulverized mycelia was purified using a phenol-chloroform extraction method followed by alcohol precipitation and evaporation, suspending the DNA in 200$\mu$l TE [@sambrook1989molecular].
Suspended DNA was stored at $4~^{\circ}$C until genotyping.

We genotyped each *S. sclerotiorum* isolate using 16 microsatellite primer pairs developed previously [@sirjusingh2001characterisation].
PCR was carried out as described previously, using primers labeled with FAM fluorophore.
Resulting amplicons were first resolved in a 1.5% agarose gel stained with ethidium bromide to ensure product was within the expected size range prior to capillary electrophoresis.
Capillary electrophoresis (fragment analysis) of amplicons, with size standard GeneScan™ 500 LIZ®, was performed using an ABI 3730 genetic analyzer (Life Technologies Corporation, Carlsbad, CA) at the Michigan State University Genomic Sequencing Center (East Lansing, MI). 
Alleles were scored using PeakScanner version 1.0 (Life Technologies Corporation, Carlsbad, CA) and recorded manually in an excel sheet.

## Data processing and Analysis{-}

<!-- All analyses were performed on this data set unless otherwise mentioned. -->
All data processing and analyses were performed in `r R.version.string %>% strsplit("\\(") %>% map_chr(1)`[@R] and are openly available and reproducible at https://github.com/everhartlab/sclerotinia-366/.
Of the 16 microsatellite loci genotyped, five included compound repeats, which made it challenging to accurately/confidently bin alleles into fragment sizes expected for each locus based on the described repeat motif.
Loci with compound repeats were removed for the reported statistics.
To ensure the integrity of the results we additionally processed these loci and included them in concurrent analyses.
We assessed the power of our 11 markers by generating a genotype accumulation curve in the R package *poppr* version `r packageVersion("poppr")`, looking for evidence of saturation, which would indicate that loci were sufficiently sampled to adequately represent the full set of haplotypes [@arnaud2007standardizing; @kamvar2015novel].
To avoid including isolates potentially collected from the same plant (which increases the probability of collecting sclerotia from the same point of infection more than once), data were clone-corrected on a hierarchy of Region/Source/Host/Year---meaning that duplicated genotypes were reduced to a single observation when they were collected in the same year from the same host cultivar located in the same source field (wmn or producer)---for subsequent analysis. 
We assessed haplotype diversity by calculating Stoddart and Taylor's index (*G*) [@stoddart1988genotypic], Shannon's index (*H*) [@shannon2001mathematical], Simpson's index ($\lambda$) [@simpson1949measurement], evenness ($E_5$), and the expected number of multilocus haplotypes (*eMLH*) [@heck1975explicit; @hurlbert1971nonconcept; @pielou1975ecological; @grunwald2003analysis]. 
To assess the potential for random mating, we tested for linkage disequilibrium with the index of association, $I_A$ and its standardized version, $\bar{r}_d$ using 999 permutations [@agapow2001indices; @smith1993how; @brown1980multilocus].
Both haplotype diversity and linkage disequilibrium were calculated in *poppr* [@kamvar2014poppr].

## Assessing Importance of Variables  {-}

### Distance-based Redundancy Analysis {-}

A distance-based redundancy analysis (dbRDA) [@legendre1999distance] was performed with the function `capscale()` in the *vegan* package version `r packageVersion("vegan")` [@vegan].
This method uses constrained ordinations on a distance matrix representing the response variable to delineate relative contribution of any number of independent explanatory variables. 
We used this method to delineate the phenotypic (Aggressiveness, Mycelial Compatibility Group (MCG)), geographic (Region, Host, Location), and temporal (Year) components in predicting genetic composition of the populations.
The distance matrix we used as the response variable was generated using Bruvo's genetic distance from clone-corrected data (procedure described above) as implemented in *poppr*, which employed a stepwise mutation model for microsatellite data [@bruvo2004simple; @kamvar2014poppr]. 
Because aggressiveness measures differed between isolates that were reduced to a single observation during clone-correction, aggressiveness was first averaged across clone-corrected isolates.
To identify explanatory variable(s) correlated with genetic variation, a forward-backward selection process was applied with the vegan function `ordistep()`.
An analysis of variance (ANOVA) was then performed to test for significance of the reduced model and marginal effects using 999 permutations.
The `varpart()` function of vegan was used to determine variation partitioning of explanatory variables.

### Aggressiveness Assessment {-}

We used ANOVA to assess if aggressiveness was significantly different with respect to Region, MCG, or multilocus haplotype (MLH). 
To minimize complications due to small sample sizes, we chose the top 10 MCGs, representing 56.5% of the isolates collected, the 10 most abundant MLHs representing 26.7% of the isolates, and populations with more than five isolates.
If ANOVA results were significantly different at $\alpha$ = 0.05, pairwise differences were assessed using Tukey's HSD test ($\alpha$ = 0.05) using the `HSD.test()` function in the package *agricolae* version `r packageVersion("agricolae")` [@mendiburu2015agricolae].

### Correlating Multilocus Haplotypes with Mycelial Compatibility Groups {-}

We wanted to assess if there was correlation between MLHs and MCGs. 
This was performed using a network-based approach where both MLHs and MCGs were considered nodes and the number of isolates in which they were found together was the strength of the connection between an MLH and and MCG node.
The network-based approach allowed us to assess how "clumpy" the relationships between MLHs and MCGs were.
To construct the network, a contingency table was created with MLHs and MCGs and converted to a directed and weighted edgelist where each edge represented a connection from an MCG to an MLH, weighted by the number of samples shared in the connection. 
This was then converted to a bipartite graph where top nodes represented MLHs and bottom nodes represented MCGs.
To identify clusters of MLHs and MCGs within the network, we used the cluster walktrap community detection algorithm as implemented in the `cluster_walktrap()` function in *igraph* version `r packageVersion("igraph")` [@csardi2006igraph; @pons2006computing].
This algorithm attempts to define clusters of nodes by starting at a random nodes and performing short, random "walks" along the edges between nodes, assuming that these walks would stay within clusters.
For this analysis, we set the number of steps within a walk to four and allowed the algorithm to use the edge weights in determining the path.
All of the resulting communities that had fewer than 10 members were then consolidated into one.
Community definitions were used to assess the average genetic distance (as defined by Bruvo's distance) within members of the community [@bruvo2004simple].

## Genetic Diversity {-}

### Population Differentiation {-}

We used analysis of molecular variance (AMOVA) with Bruvo's genetic distance in *poppr* to test for differentiation between populations in wmn and producer fields from the same region and collected in the same year [@excoffier1992analysis; @bruvo2004simple; @kamvar2014poppr].
To identify Regions with greater differentiation, we used discriminant analysis of principal components (DAPC) as implemented in *adegenet* version `r packageVersion("adegenet")`, assessing the per-sample posterior group assignment probability [@jombart2008adegenet]. 
This method decomposes the genetic data into principal components, and then uses a subset of these as the inputs for discriminant analysis, which attempts to minimize within-group variation and maximize among-group variation [@jombart2010discriminant]. 
To avoid over-fitting data, the optimal number of principal components was selected by using the *adegenet* function `xvalDapc()`. 
This function implements a cross-validation procedure to iterate over an increasing number of principal components on a subset (90%) of the data, trying to find the minimum number of principal components that maximizes the rate of successful group reassignment. 
To assess if cultivar had an influence on genetic diversity between wmn, we first subset the clone-corrected data to contain only samples from wmn and from the cultivars Beryl, Bunsi, and G122 and tested differentiation using AMOVA and DAPC as described above. 
We additionally assessed population stability over time by calculating DAPC over the combined groups of Region and Year as described above.

### Analysis of Shared Multilocus Haplotypes {-}

We wanted to evaluate patterns of connectivity between shared multilocus haplotypes across geographic regions.
We first tabulated the multilocus haplotypes shared between at least two populations (defined as states or countries) with the *poppr* function `mlg.crosspop()` [@kamvar2014poppr].
From these data, we constructed a graph with populations as nodes and shared haplotypes as edges (connections) between nodes using the R packages *igraph* [@csardi2006igraph], *dplyr* version `r packageVersion("dplyr")` [@dplyr], and *purrr* version `r packageVersion("purrr")` [@purrr].
Each node was weighted by the fraction of shared MLHs.
Each edge represented a single MLH, but because a single MLH could be present in more than one population, that MLH would have a number of edges equivalent to the total number of possible connections, calculated as (*n*\*(*n*-1))/2 edges where n represents the number of populations crossed.
Edges were weighted by $1-P_{sex}$, where $P_{sex}$ is the probability of encountering the same haplotype via two independent meiotic events [@arnaud2007standardizing; @parks1993study].
This weighting scheme would thus strengthen the connection of edges that represented genotypes with a low probability of being produced via sexual reproduction. 
We then identified communities (among the Regions) in the graph using the `cluster_optimal()` function from *igraph* [@csardi2006igraph].
The graph was plotted using the R packages *ggplot2* version `r packageVersion("ggplot2")` [@ggplot2] and *ggraph* `r packageVersion("ggraph")` [@ggraph].
To ensure that we captured the same community signal, we additionally performed this analysis including the five polymorphic markers described above.

# Results {-}

A total of 366 isolates were collected from 2003 to 2012 (except 2006 and 2011) from diseased dry bean plants in Australia, Mexico, France, and 11 states in the USA (Table \@ref(tab:isolate-table)).
Over 50% of the isolates came from four states, MI (`r pop_size[["MI"]]`), ND (`r pop_size[["ND"]]`), WA (`r pop_size[["WA"]]`), NE (`r pop_size[["NE"]]`). 
Four regions had fewer than 10 isolates, Australia (`r pop_size[["Australia"]]`), WI (`r pop_size[["WI"]]`), NY (`r pop_size[["NY"]]`), ID (`r pop_size[["ID"]]`).
We observed 87 MCGs, the most abundant of which ('MCG 5') was represented by 73 isolates over 37 MLHs (Fig \@ref(biggraph)A,C).
Analysis of the haplotype accumulation curve showed no clear plateau (Supplemental Information), indicating that we would likely obtain more multilocus haplotypes if we were to haplotype more loci. The number of observed alleles per locus ranged from two to 10 with an average of `r signif(Amean, 3)` (Table \@ref(tab:locus-stats)).
After clone-correction on the hierarchy of Region/Source/Host/Year, we were left with `r nInd(dat11cc)` isolates representing `r nmll(dat11cc)` MLHs (Table \@ref(tab:iatab)).


```{r , echo = FALSE, warning = FALSE, message = FALSE, results = "asis"}
loc_table %>%
  mutate(allele = round(allele, 2)) %>%
  mutate(Repeats = case_when(
    Repeats <= 2 ~ "di-",
    Repeats <= 3 ~ "tri-",
    Repeats <= 4 ~ "tetra-",
    Repeats <= 5 ~ "penta-",
    Repeats <= 6 ~ "hexa-"
  )) %>%
  rename(`*h*` = Hexp) %>%
  rename(`No. alleles` = allele) %>%
  rename(`Repeat Motif` = Repeats) %>%
  huxtable::as_huxtable(add_colnames = TRUE) %>%
  huxtable::set_number_format(huxtable::everywhere, 3:4, 0) %>%
  huxtable::set_number_format(huxtable::everywhere, 5:6, 2) %>%
  huxtable::set_align(huxtable::everywhere, 2:6, "right") %>%
  huxtable::set_col_width(c(0.06, 0.07, 0.11, 0.1, 0.05, 0.075)) %>%
  huxtable::print_md(max_width = 61)
```

Table: (\#tab:locus-stats) Allelic diversity on full data set at loci used in this study. *h* = Nei's Gene Diversity [@nei1978estimation]. Average *h* = `r signif(Hmean, 3)`, average Evenness = `r signif(Emean, 3)`, average no. alleles = `r signif(Amean, 3)`

```{r table_ia, echo = FALSE, warning = FALSE, message = FALSE, results = "asis"}
suppressPackageStartupMessages(library("tidyverse"))
read_csv("../../results/tables/genotype_diversity_table_region.csv", 
         na = "NA", # spaces read in as spaces
         col_types = cols(
         Pop = col_character(),
         N = col_integer(),
         MLG = col_integer(),
         eMLG = col_double(),
         SE = col_double(),
         H = col_double(),
         G = col_double(),
         lambda = col_double(),
         E.5 = col_double(),
         Hexp = col_double(),
         Ia = col_double(),
         p.Ia = col_double(),
         rbarD = col_double(),
         p.rD = col_double(),
         File = col_character()
      )) %>%
  # mutate(Ia = case_when(
  #   !is.finite(p.Ia) ~ "-",
  #   p.Ia == 0.001 ~ paste0(round(Ia, 2), "*"),
  #   p.Ia <= 0.01  ~ paste0(round(Ia, 2), "-"),
  #   p.Ia <= 0.05  ~ paste0(round(Ia, 2), "~"),
  #   TRUE          ~ as.character(round(Ia, 2))
  # )) %>%
  mutate(rbarD = case_when(
    !is.finite(p.rD) ~ "-",
    p.rD == 0.001 ~ paste0(round(rbarD, 2), "*"),
  #   p.rD <= 0.01  ~ paste0(round(rbarD, 2), "-"),
  #   p.rD <= 0.05  ~ paste0(round(rbarD, 2), "~"),
    TRUE          ~ as.character(round(rbarD, 2))
  )) %>%
  mutate_if(is.numeric, round, digits = 2) %>%
  mutate(N = paste(N, MLG, sep = " (") %>% paste0(")") %>% formatC()) %>%
  mutate(eMLG = paste(eMLG, SE, sep = " (") %>%  paste0(")") %>% formatC()) %>%
  rename(`*h*` = Hexp) %>%
  rename(`$E_5$` = E.5) %>%
  # rename(`$I_A$` = Ia) %>%
  rename(`$\\lambda$` = lambda) %>%
  rename(`$\\bar{r}_d$` = rbarD) %>%
  rename(eMLH = eMLG) %>%
  select(-File, -MLG, -SE, -p.Ia, -p.rD, -Ia) %>% 
  huxtable::as_huxtable(add_colnames = TRUE) %>%
  huxtable::set_number_format(huxtable::everywhere, 4:5, 1) %>%
  huxtable::set_align(huxtable::everywhere, -1, "right") %>%
  huxtable::set_col_width(c(1.1, 0.9, 1.3, 0.5, 0.5, 1.1, 0.7, 0.6, 0.7)) %>%
  huxtable::print_md(max_width = 80)
```

Table: (\#tab:iatab) Genotypic diversity and Linkage Disequilibrium summary for geographic populations arranged by abundance after clone-correction by a hierarchy of Region/Source/Host/Year. 
Pop = Population, N = number of individuals (number of MLH in parentheses), 
eMLH = expected number of MLHs based on rarefaction at min(10, N) individuals (standard error in parentheses), 
H = Shannon-Weiner Index, G = Stoddardt and Taylor's Index, 
$\lambda$ = Simpson's Index, 
*h* = Nei's 1978 gene diversity, 
$E_5$ = Evenness,
$\bar{r}_d$ = standardized index of association.
An asterix indicates a significant value of $\bar{r}_d$ after 999 permutations, $P \leq 0.001$.

<!-- MCG graph figure -->

![
Data show complex associations between Mycelial Compatibility Groups and Multilocus Haplotypes.
**A)** Barplot of Mycelial Compatibility Group (MCG) abundance in descending order.
Singletons (46) were truncated, leaving 41 MCGs.
White bars represent sample counts and grey bars represent counts of unique multilocus haplotypes (MLH).
The transparency of the bars represent the evenness of the distribution of the MLHs within a given MCG.
A dashed box surrounds the 8 most common MCGs representing > 51% of the data.
**B)** Full graph-representation of the relationship between MCGs (open circles) and MLHs (filled circles).
Details in fig. \@ref(fullgraph). 
**C)** A subset of **B** representing the 8 most common MCGs and their associated MLHs (dashed box in **A**).
Filled nodes represent MLHs and open nodes represent MCGs.
Numbers inside nodes are the MLH/MCG label. 
Node (circle) area scaled to the number of samples represented (range: 1--73).
Edges (arrows) point from MLH to MCG where the weight (thickness) of the edge represents the number of shared isolates (range: 1--19).
Edges extending from MLHs displayed to other MCGs are not shown.
](../../results/figures/publication/mcg-subgraph-with-context.pdf){#biggraph width=100%}

<!-- end fig -->

## Variable Assessment {-}

### Variable Importance {-}

The forward-backward selection process of the dbRDA models on clone-corrected data revealed `r RDAVAR1` to be the optimal variables for the reduced model, accounting for `r prettify(vartab(caplist1$components)[2, "Proportion"])`% of the total variation.
ANOVA revealed that the reduced model was significant with an adjusted R^2^ of `r RDA1R2` (*P* = 0.001).
Assessment of the marginal effects showed that all of these varaibles significantly explained genetic variation (*P* = `r anova1res[["Year"]]`, *P* = `r anova1res[["Region"]]` for region).
It was revealed that there was multicollinearity when MCG was combined with any other variable, so we followed the same procedure, dropping MCG from the list of potential predictors. 
From these results, we found that `r RDAVAR2` were optimal, accounting for `r prettify(vartab(caplist2$components)[2, "Proportion"])`% of the total variation.
ANOVA revealed significant effects with an adjusted R^2^ of `r RDA2R2` (*P* = 0.001).
While the marginal effect assessment showed that `r andlist(names(anova2res)[anova2res <= 0.01])` all significantly explained the variation at *P* = `r anova2res[["Year"]]`, and Aggressiveness significantly explained the variation at *P* = `r anova2res[["Severity"]]`.
Much of the variation appears to be driven by isolates from Mexico and 2005 (Fig. \@ref(rda-biplot)).
Variance partitioning of the independent variables without MCG indicated aggressiveness to be the least influential factor with 0.1% contributing to explaining the variation of molecular data whereas the combination of variables accounted for 3.3%.

<!-- RDA figure -->

![
Biplot showing five most influential explanatory variables (arrows) overlayed on the first two eigenvectors of distance based redundancy analysis of *S. sclerotiorum* isolates. 
The length of the arrows are directly proportional to the strength of the correlation between explanatory and molecular variables. 
Open circles represent the 318 clone-corrected samples in ordination space. 
](../../results/figures/publication/Figure7Z.pdf){#rda-biplot width=61%}

<!-- end figure -->

### Aggressiveness {-}

<!-- aggressiveness figure -->

![
Boxplot representing aggressiveness (vertical axis) by population (horizontal axis) arranged in descending order of mean aggressiveness for all populations with N > 5. Box boundaries represent 25% and 75% percentiles and whiskers extend to at most 1.5 $\times$ the interquartile range. Horizontal line represents median value. Circles represent individual observations where filled circles indicate isolates from white mold screening nurseries (wmn).
](../../results/figures/publication/aggressiveness.pdf){#aggressiveness-boxplot}

<!-- end figure -->

Our assessment of aggressiveness in association with Region showed a significant effect (P < 1.00e^-4^), with means that ranged from 5.8 (MN) to 4.0 (CA) (Fig. \@ref(aggressiveness-boxplot), Table \@ref(tab:region-aggressiveness)). 
MCGs also showed a significant effect (P < 1.00e^-4^), with means that ranged from 6.0 ('MCG 44') to 4.6 ('MCG 49') (Table \@ref(tab:MCG-aggressiveness)).
We additionally found a significant effect for MLHs (*P* = 7.44e^-4^) for MLHs, with means that ranged from 6.0 ('MLH 78') to 4.3 ('MLH 140') (Table \@ref(tab:MLG-aggressiveness)).

### Correlation of Mulitlocus Haplotypes and Mycelial Compatibility Groups {-}

Our network-based approach to correlating MLHs with MCGs revealed a large and complex network (Fig. \@ref(biggraph)). 
Community analysis showed 51 communities, `r sum(in_reach$N == 0)` of which consisted of a single MLH unconnected with any other community indicating that just `r prettify(mean(in_reach$N == 0))`% of the 165 MLHs are unable to cross with any other MLH in this data set (Fig. \@ref(fullgraph)).
The three communities with the most members contained eight of the 10 most abundant MCGs. 
Comparing these communities with Bruvo's genetic distance revealed an average distance of 0.451 among communities and an average distance of 0.437 within communities, showing no significant difference.
When we assessed the number of times two different MLHs that are in the same MCG, considering these as potential heterothallic pairings that could result in sexual recombination, we found an average of `r signif(mean(in_reach$N), 3)` potential heterothallic parings per MLH. 
Representing just four isolates, 'MLH 75' had `r big_haplotype$N` neighbors that shared the same mycelial compatibility group (Fig \@ref(biggraph), \@ref(fullgraph)). 
Overall, there was no clear pattern to the association between MLH and MCGs. 

## Structure of Shared Multilocus Haplotypes {-}

We observed a total of 165 MLHs (215 with 16 loci).
The most abundant MLH was represented by 27 isolates (Table \@ref(tab:mlg-table)) from five regions (NE, MI, WA, CO, and ND). 
Within regions, the haplotypes were relatively evenly distributed with moderate to high diversity (Table \@ref(tab:iatab)). 
Of the 165 MLHs, 76 (46%) were found in at least two geographic regions except those found in WI (2), ID (1), and Mexico (18) (Fig. \@ref(community-graph)).
Community analysis of the MLHs shared between populations revealed 4 communities with a modularity of 0.17 (Fig. \@ref(community-graph)): A coastal community (CA, OR, WA, and NY), a midwest community (CO, NE, ND, MI), and an international community (MN, France, Australia).
Analysis with 16 loci resulted in the removal of the NY node because it no longer shared a haplotype with OR, but the same overall community structure was present with a modularity of 0.2 (Fig. \@ref(community-graph16)).
The international community appears to be driven by MLH 4, which is shared between all three populations and has a $P_{sex}$ value of 2.87e^-5^, in contrast to the abundant MLH 25, which has a $P_{sex}$ value of 0.0168.


<!-- community graph figure -->

![
Network of populations (nodes/circles) and their shared multilocus haplotypes (MLH) (edges/lines) haplotyped over 11 loci. 
Each node is labeled with **name (number of MLHs shared/number of MLHs total).** 
The shade and area of the nodes are proportional to the number of unique MLHs within the node and the inner nodes are proportional to the number of private MLHs to the region (bottom legend). 
Each edge represents a single MLH where its thickness represents the number of populations that share the MLH and the shade represents the value of $P_{sex}$, or the probability of encountering that MLH from two independent meiotic events.
](../../results/figures/publication/mlg-11.pdf){#community-graph width=75%} 

<!-- end fig -->


```{r table3, echo = FALSE, warning = FALSE, message = FALSE, results = "asis"}
read_csv("../../results/tables/mlg_table.csv", 
         na = "NA", # spaces read in as spaces
         col_types = cols(.default = col_character())) %>%
  rename(`$P_{sex}$` = Psex) %>% 
  rename(MLH = MLG) %>%
  knitr::kable(format = "pandoc")
```

Table: (\#tab:mlg-table) The 5 most abundant Multilocus Haplotypes (MLH) with
the probability of second encounter ($P_{sex}$), Mycelial Compatibility Groups (MCG), and Regions with sample sizes in parentheses.




## Population Differentiation {-}

### Analysis of Molecular Variance {-}

The analysis of molecular variance (AMOVA) over the hierarchy of Region, Source, and Year showed significant variation between regions and years, but no significant variation between wmn and non-wmn (Table \@ref(tab:amova)).
In contrast, when we compared the three cultivars, Beryl, Bunsi, and G122, we found no significant differentiation (Supplementary Information).

```{r table4, echo = FALSE, warning = FALSE, message = FALSE, results = "asis"}
read_csv("../../results/tables/AMOVA-year.csv", 
         na = "NA", # spaces read in as spaces
         col_types = cols(.default = col_character()))  %>%
  rename(ps = `Phi statistic`) %>%
  select(-Sigma) %>%
  # mutate_if(is.numeric, format, digits = 3) %>%
  mutate(ps = signif(as.numeric(ps), digits = 3)) %>%
  mutate(ps = gsub("0\\.00(\\d{1})(\\d{2})", "\\1.\\2e^-3^", ps)) %>%
  mutate(ps = case_when(P > 0.05 ~ ps, TRUE ~ paste0("**", ps, "**"))) %>%
  rename(`$\\Phi$ statistic` = ps) %>%
  rename(`% variation` = `Percent variation`) %>%
  rename(S.S. = `Sum of Squares`) %>%
  rename(`*P*` = P) %>%
  select(c(1:4, 6, 5)) %>%
  huxtable::as_huxtable(add_colnames = TRUE) %>% 
  huxtable::set_col_width(c(0.8, 0.15, 0.15, 0.35, 0.5, 0.15)) %>% 
  huxtable::set_align(huxtable::everywhere, 2:6, "right") %>% 
  huxtable::set_number_format(huxtable::everywhere, 2, 0) %>%
  huxtable::set_number_format(huxtable::everywhere, 6, 3) %>%
  huxtable::set_number_format(huxtable::everywhere, 5, 4) %>%
  huxtable::print_md(max_width = 80)
```

Table: (\#tab:amova) Analysis of Molecular Variance on Bruvo's genetic distance showing little differentiation between white mold screening nurseries (wmn) and other sources. The hierarchy was constructed as Source/Region where source is defined as belonging to a wmn or not. Bold $\Phi$ values indicate significant values (*P* < 0.05). S.S. = Sum of Squares, d.f. = degrees of freedom.

### Discriminant Analysis of Principal Components {-}

Discriminant Analysis of Principal Components (DAPC) was performed grouping by Region with the first `r dapc_region$n.pca` principal components, representing `r prettify(dapc_region$var)`% of the total variance. 
The first discriminant axis (representing `r paxis(dapc_region)`% of the discriminatory power) separated the centroid for the Mexico isolates from the rest of the data, indicating strong differentiation (Fig. \@ref(DAPC)b). 
The second discriminant axis, representing `r paxis(dapc_region, 2)`% of the discriminatory power, separated the centroid for the California isolates.
The mean population assignment probabilities for all populations with n > 10 showed that only isolates from Mexico, CA, and France had > 50% probabilities of being reassigned to their source populations (Fig. \@ref(DAPC)a).

<!-- DAPC figure -->

![
Discriminant Analysis of Principal Components on regions showing that Mexico is differentiated from other populations. 
**A)** Mean population assignment (horizontal axis) probability (vertical axis) for all populations with N > 10 (facets).
Numbers next to source populations indicate population size.
All values sum to one.
**B)** Scatter plot of first two components from DAPC.
Points represent observed individuals connected to the population centroids with ellipses representing a 66% confidence interval for a normal distribution. 
The center of each component is represented as black grid lines.
](../../results/figures/publication/DAPC.pdf){#DAPC width=100%}

<!-- end figure -->

DAPC grouping by cultivar used the first 20 principal components, representing 89% of the total variance.
The first two discriminant axes (representing `r sum(paxis(dapc_host, 1:2))`% of the discriminatory power) failed to separate any of the cultivars where the mean posterior assignment probabilities were `r passp(dapc_host, "G122")`% (G122), `r passp(dapc_host, "Beryl")`% (Beryl), and `r passp(dapc_host, "Bunsi")`% (Bunsi).
DAPC grouping by Region and Year used the first `r dapc_region_year$n.pca` principal components, representing `r prettify(dapc_region_year$var)`% of the total variance.
The Midwestern populations (NE, MI, CO, ND) did not appear to have any variation across time in contrast to WA, which showed a shift in population structure in the last year of sampling, 2008 (Fig. \@ref(DAPC-RY)).
Further analysis of this population revealed that all 12 isolates in WA circa 2008 originated in a wmn and nine of these haplotypes were shared with CA, and three were shared with France (Fig. \@ref(community-graph), \@ref(community-graph16)).

<!-- DAPC figure 2: Electric Boogaloo-->

![
Scatter plot of Discriminant Analysis of Principal Components on Regions and Years showing temporal variation in WA and CA populations as compared to Midwestern Regions, NE and MI. 
Points (text labels) represent observed individuals connected to the population centroids with ellipses representing a 66% confidence interval for a normal distribution. 
The center of each component is represented as black grid lines.
A more detailed view can be seen in Fig. \@ref(DAPC-RY-FULL).
](../../results/figures/publication/dapc_region_year_micanewa.pdf){#DAPC-RY width=61%}

<!-- end figure -->


### Linkage Analysis {-}

Nearly all populations showed evidence for linkage (Table \@ref(tab:iatab)), which serves as evidence for clonal reproduction or other forms of non-random mating. 
The only exceptions are CA and Australia. Both of these populations showed only moderate significance (P < 0.05) with $\bar{r}_d$ values of 0.03 and 0.12, respectively. 

# Discussion {-}
 
In this study, we characterized the diversity of *Sclerotinia sclerotiorum* from dry bean fields across the United States.
Our results suggest that, broadly, populations from white mold screening nurseries reflect the populations of the surrounding regions, indicating that resistance screening may be successful at a within regions. 
We found significant population differentiation by geographic region and year, mainly differentiated into three broad North American groups based on shared haplotypes and posterior groupings, a Coastal Region, Midwestern Region, and Mexico. 
To date, with 366 isolates, this is the largest single population genetic study of *S. sclerotiorum* assessing population structure within managed and unmanaged agricultural environments. 
These findings indicate that the wmn can be effective at screening for potential resistant lines within growing regions. 

## Variable Importance {-}

Borrowing a technique often used in the ecological literature, we used dbRDA to elucidate the effect of all variables (`r andlist(c(names(strata(dat11)), "Aggressiveness"))`) [@legendre1999distance].
From the initial results, it appears that the most important factors for predicting genetic structure are MCG, region, and year.
When we inspected the biplot of the initial results, we saw that the most important predictors were 'MCG 44', 'MCG 5', and 'MCG 9'. 
We believe that this is driven by the fact that these particular MCGs have uneven MLH distributions, meaning that they are heavily associated with one particular MLH (Fig. \@ref(biggraph)).
We note these results with caution because of the apparent multicolinearity between MCG and Region, which is a violation of the analysis [@legendre1999distance].
While the results indicated that Mexico and the year 2005 were the two most important variables, it's worth noting that all Mexico isolates were collected in 2005 (Fig. \@ref(rda-biplot)).
The results also show that the Vista cultivar explains some of the variance, but this represents six isolates in MI. 
Aggressiveness and specific source field had little to no effect on genetic diversity. 
These results are in agreement with studies that examined differentiation based on Host [@aldrich-wolfe2015genetic] and Aggressiveness [@atallah2004high; @attanayake2012genetic; @attanayake2013sclerotinia] reporting little or no correlation of genetic diversity to these variables. 
This indicates that a) breeders should keep in mind regional differences when assessing resistance and b) it is possible that we have not yet measured biologically relevant variables that can predict genetic differentiation, which could include variables such as soil community composition. 

## Mycelial Compatibility Groups and Multilocus Haplotypes {-}

Of interest in this study was testing the association between multilocus haplotypes (MLHs) and mycelial compatibility groups (MCGs), as the latter have historically been used as a proxy for clonal lineages [@leslie1993fungal; @otto-hanson2011variation; @kohn1990mycelial; @schafer2006optimized; @kohn1995clonal; @carbone1999patterns]. 
Our results do not support this assumption.
It can be seen in Fig. \@ref(biggraph)A that the most abundant MCG contains several MLHs, but the diversity of those MLHs are low as indicated by the evenness (transparency), which indicates that there is one dominant MLH ('MLH 25').
What is not shown in Fig. \@ref(biggraph)A is the MLHs that are shared between MCGs. 
This is illustrated in both table \@ref(tab:mlg-table) and Fig. \@ref(biggraph)B,C. 
It could be argued, however that 'MLH 25', with its high value of $P_{sex}$ represents different true MLHs across the five MCGs it occupies, but this doesn't account for the overall structure of Fig. \@ref(fullgraph) where, for example, 'MLH 75' ($P_{sex}$ = 1.81e^-4^) is compatible with `r big_haplotype$N` other haplotypes through three MCG when the population structure of *S. sclerotiorum* is known to be clonal.

Over the past few years, researchers have noticed inconsistencies among the relationship between MCGs and MLHs [@aldrich-wolfe2015genetic; @carbone1999patterns; @attanayake2012genetic; @lehner2015genetic]. 
Either several MCGs belong to one MLH, which could be explained by insufficient sampling of loci; several MLHs belong to one MCG, which could be explained by clonal expansion; or a mixture of both.
Some studies have shown a correlation between MCG and MLH [@aldrich-wolfe2015genetic; @carbone1999patterns; @lehner2015genetic], whereas other studies have shown no apparent correlation, even on small spatial scales [@atallah2004high; @attanayake2012genetic; @attanayake2013sclerotinia].

One long-held assumption was that MCGs (as determined via barrage reaction) represent vegetative compatibility groups (VCGs) [@schafer2006optimized; @kohn1990mycelial; @lehner2015genetic], which are known to have a genetic component [@saupe2000molecular; @hall2010evolution; @strom2016genomes].
While our protocol for assessing MCGs utilized a different substrate for the MCG reactions, the patterns we observe are not dissimilar from what have previously been reported in the literature.
It has been demonstrated in several Ascomycetes---including *Neurospora crassa* [@micali2003independence], *Sclerotinia homoecarpa* [@jo2008reassessment], *Verticillium dahliae* [@papaioannou2014barrage], and *S. sclerotiorum* [@ford1995heterokaryon]---that barrage reactions are independent from stable anastomosis.
Thus, the inconsistencies in this study and other studies indicate that researchers studying *S. sclerotiorum* should not rely on MCG data derived from barrage reactions as an indicator for genetic diversity.  
 
## Genotypic Structure {-}

The results show several haplotypes shared across populations, except for Mexico and two states that had fewer than three samples each (Fig. \@ref(community-graph)).
Our network-based approach by treating the haplotypes as edges and weighting each edge with the inverse of $P_{sex}$ treated the edges as springs connecting the populations with the strength proportional to the probability of obtaining the same haplotype as a clone. 
This allowed us to use a graph walking algorithm to see how close the populations were simply based off of the proportion of clones they shared.
The most abundant haplotype was shared across four populations, but its high value of $P_{sex}$ meant that it did not contribute significantly to the overall structure. 
The graph walking algorithm was able to divide the network into three groups, but had a modularity of 0.17, which indicates that the groups are only weakly differentiated.

The widespread nature of multilocus haplotypes in both wmn and production fields with relatively small values of $P_{sex}$ may indicate the spread of inoculum between regions. 
It has previously been shown that infections can be transmitted through seed [@botelho2013performance], but seedborne transmission is thought to be of insignificant epidemiological importance [@strausbaugh2003management].
Thus, we hypothesize that shared haplotypes between populations may arise due to transmission events of seed or sclerotia.
However, because we observe population structure based on Region, we speculate that these transmission events are rare. 

## Population Differentiation {-}

The primary interest of this study was to assess if isolates sampled from white mold screening nurseries represent isolates from producer fields within the region [@steadman2003identification; @otto-hanson2011variation].
According to our AMOVA results, we have evidence for differentiation at the Region and Year, but little to no differentiation between wmn isolates and production field isolates (Table \@ref(tab:amova)).
When we analyze the AMOVA results in light of the DAPC results (Fig. \@ref(DAPC)), it becomes clear that the regional patterns of differentiation are largely driven by isolates from Mexico and CA.
Isolates from these Regions had a higher posterior probability (> 0.75) of being reassigned to their own populations than any other (Fig. \@ref(DAPC)A). 
All other populations in comparison (except France) has reassignment probabilities of < 0.5, which is reflected in the failure of the first two discriminant functions to separate these populations (Fig. \@ref(DAPC)B).

Despite the evidence that Mexico and CA contributed to much of the population differentiation, Regions like WA still had a large amount of internal variation.
The two distinct clusters for the WA Region showed that the 2008 population appeared differentiated and, under further investigation, we found that all the haplotypes from this year were shared between CA and France (Fig. \@ref(community-graph), \@ref(DAPC-RY), \@ref(DAPC-RY-FULL)). 
All of the isolates from WA in 2003--2005, and 2008 came from the same wmn; those in 2003--2005 came a NE field location cropped with dry bean since 2002, and those in 2008 from a SE field that was previously cropped with brassica, sundgrass, peas, beans, and potatoes (Miklas, Phil Pers. comm.).
Both of these fields were inoculated with sclerotia in 2002, the NE field with sclerotia provided by a commercial bean producer and the SE field with sclerotia from peas (although this was thought to be unsuccessful).
Despite this information, it is still unclear what has contributed to the differentiation of the 2008 population from WA.
While the evidence my suggest host as being a factor, previous studies have shown no significant differentiation across host species [@aldrich-wolfe2015genetic]. 

With the exception of the WA Region, populations that were sampled across several years appeared to be relatively stable over time with overlapping distributions in the DAPC (i.e. NE and MI, Fig. \@ref(DAPC-RY)). 
DAPC is based on the principal components of Euclidean distance between genotypes [@jombart2010discriminant]. 
Unlike Bruvo's distance, Euclidean distance does not take into account the magnitude of the difference between alleles, which could inflate the distance measure in the presence of private alleles [@bruvo2004simple]. 
We found no evidence of private alleles in the Mexico and CA isolates, suggesting that this pattern separating these populations from the others was not an artifact.

It has been proposed within the *S. sclerotiorum* literature that isolates from sub-tropical and tropical climates are differentiated or more variable than populations from temperate climates [@lehner2017sclerotinia;@carbone2001multilocus; @attanayake2013sclerotinia]. 
This has been attributed to the notion that the fungus has the chance to undergo more reproductive cycles in the warmer climate [@carbone2001multilocus; @attanayake2013sclerotinia]. 
Populations in sub-tropical regions of China have been found to be more variable, sexually reproducing, and unrelated to populations in temperate regions of the USA [@attanayake2013sclerotinia].
The differentiation between the Mexico isolates and all other regions gives some evidence for this hypothesis as the only isolates in our study from sub-tropical regions are in Mexico. 

We were unable to find evidence for sexual reproduction in any region except for CA and Australia. 
While the Australia population had a non-significant value of $\bar{r}_d$---which would suggest that we cannot reject the null hypothesis of random mating---the sample size was insufficient from which to draw conclusions [@agapow2001indices; @milgroom1996recombination].
The low value of $\bar{r}_d$ in the CA population may represent sexual reproduction, but this could also be an artifact of sampling two different populations, which is known to reduce the value of $\bar{r}_d$ [@prugnolle2010apparent].


## Aggressiveness {-}

We observed significant differences in aggressiveness based on region (Fig. \@ref(aggressiveness-boxplot), Table \@ref(tab:region-aggressiveness)). 
These results show a similar pattern to what was found previously with the exception of North Dakota, which increased in mean aggressiveness from 5 to 5.77 [@otto-hanson2011variation].
This increase was due in part to new data from producer field isolates collected after the previous study.
Two caveats should come with interpreting these data, however.
First, these straw tests were performed by a different person for these later isolates, which could suggest a more lenient or strict scoring system. 
Moreover to this point, many of the ND isolates fell within the 6--7 range, which denotes a physical boundary (disease symptoms around the second node) between intermediate and susceptible [@otto-hanson2011variation]. 
Second, these results are all within the intermediate range (4--6) of aggressiveness, which may not represent a significant biological difference.


## Limitations {-}

One of the main limitations of this study is the focus on *P. vulgaris* as a host. 
It has been shown that *S. sclerotiorum* in the midwestern United States does not have a particular preference for host [@aldrich-wolfe2015genetic]. 
If the distribution of *S. sclerotiorum* is even across agricultural hosts in the USA, then our sample may yet be representative of the genetic pool present in other crops and weedy species. 

Another limitation was the microsatellite markers used for this particular study [@sirjusingh2001characterisation].
The haplotype accumulation curve showed no indication of a plateau, indicating that if we had sampled more loci, we would have resolved more multilocus haplotypes. 
While 16 loci showed us similar results and began to show a plateau for the haplotype accumulation curve, we were unable to use these results due to our uncertainty in the allele calls for these five extra loci. 
With the availability of an optically-mapped genome [@derbyshire2017complete], future studies describing the genetic diversity of *S. sclerotiorum* should employ techniques such as Genotyping-By-Sequencing [@davey2011genome], Sequence Capture [@grover2012targeted], or Whole Genome Sequencing.

## Conclusions {-}

This study represents the largest genetic analysis of *S. sclerotiorum* from the USA to date, giving us a unique insight to continent-wide population structure and relationships between phenotypic and genotypic variables.
Based on our analysis of the relationships between MCG and MLH, we found no clear evidence that the two are directly related, suggesting that MCG do not necessarily represent vegetative compatibility groups and thus should not be used as a proxy for identifying clones. 
Populations in wmn appear to show no significant differentiation when compared to their production field counterparts, suggesting that the wmn populations of *S. sclerotiorum* may be considered representative of the surrounding regions. 
While we found no direct relationship between haplotype and severity, it is evident that there is a gradient of severity by region, further supporting the need for screening in multiple locations.

## Data Availability {-}

All scripts, data, and resources used to generate the results presented in this publication (including supplementary information) are fully reproducible and available at The Open Science Framework https://osf.io/ejb5y. 

## Acknowledgements {-}

The authors would like to thank Rebecca Higgens for providing valuable insights into the historical context of the data collection and curation. Funding for this research was provided by the North Central Soybean Research Program (#639K623) to SEE, a Layman Award (#2446) to SEE, USDA-ARS National Sclerotinia Initiative (#58-5442-2-209) to JRS/SEE, and start-up funds from the University of Nebraska-Lincoln to SEE.

## Conflicts of Interest {-}

The authors declare no conflict of interest.

## Author Contributions {-}

 - **Zhian N. Kamvar** analyzed the data, contributed analysis tools, wrote the paper, prepared figures and tables, reviewed drafts of the paper.

 - **Bimal Sajeewa Amadarasa** ...

 - **Rachana Jhala** ...

 - **Serena McCoy** ...

 - **James Steadman** ...

 - **Sydney E. Everhart** ...

\newpage

# Supplementary Information {-}


<!-- This part is necessary for adding the "S" before supplementary figures.
Will not work with HTML outputs, though. -->

\setcounter{table}{0}
\renewcommand{\thetable}{S\arabic{table}}
\setcounter{figure}{0}
\renewcommand{\thefigure}{S\arabic{figure}}

-----------------------------------------------------------------------------
Country   State  Field Code              Year Host                          N 
--------- ------ ---------- ----------------- -------------------------- ----
USA       CA     wmn               2004, 2005 Beryl, Bunsi, G122           18 

USA       CO     producer          2007, 2010 Pinto, Yellow                41 

                 wmn                     2003 GH                            1 

USA       ID     producer                2003 GH                            1 

USA       MI     wmn        2003, 2004, 2005, 11A, 37, 38, B07104,         43 
                                   2008, 2009 Beryl, Bunsi, Cornell,       
                                              G122, Orion, PO7863, WM31      

                 producer    2003, 2008, 2009 BL, Black, Fuji, GH,         19 
                                              Merlot, SR06233, unk,        
                                              Vista, Zorro 

USA       MN     wmn               2003, 2004 Beryl, Bunsi, G122           11 

USA       ND     producer          2007, 2010 unk                          53 

                 wmn                     2005 Beryl, Bunsi, G122            7 

USA       NE     wmn        2004, 2005, 2008, Beryl, Bunsi, G122,          27 
                                         2010 PO7683, unk                    

                 producer   2003, 2007, 2009, Beryl, Emerson, GH, Orion,   20 
                                         2010  Pinto, Weihing                 

USA       NY     producer                2003 GH                            1 

USA       OR     wmn               2003, 2004 Beryl, Bunsi, G122           15 

                 producer                2003 G122, GH                      2 

USA       WA     wmn        2003, 2004, 2005, 11A, 37, 38, Beryl, Bunsi,   36 
                                         2008  Cornell, G122, Orion, PO7      
                                              104, PO7863, WM31               

                 producer          2003, 2007 GH, Merlot, Pinto, Redkid    23 

USA       WI     producer                2003 GH                            2 

Mexico    -      wmn                     2005 Beryl, Bunsi, G122           18 

France    -      wmn               2004, 2005 Beryl, Bunsi, G122           18 

                 producer                2012 unk                           4 

Australia -      wmn                     2004 Beryl, Bunsi, G122            4 

                 producer                2004 Beryl                         2 

-----------------------------------------------------------------------------

Table: (\#tab:isolate-table) Description of *Sclerotinia sclerotiorum* isolates used in this study.
N = Number of Isolates. 
Key abbreviations: wmn = white mold screening nursery, producer = producer field, unk = unknown cultivar.


```{r table2, echo = FALSE, warning = FALSE, message = FALSE, results = "asis"}
suppressPackageStartupMessages(library("tidyverse"))
read_csv("../../results/tables/Region-aggressiveness.csv", 
         na = "NA", # spaces read in as spaces
         col_types = cols(.default = col_character())) %>%
  huxtable::as_huxtable(add_colnames = TRUE) %>%
  huxtable::set_number_format(row = huxtable::everywhere, col = 1, 0) %>% 
  huxtable::set_col_width(c(0.13, 0.25, 0.08)) %>%
  huxtable::set_align(row = huxtable::everywhere, col = 2, "right") %>%
  huxtable::print_md(max_width = 40)
```

Table: (\#tab:region-aggressiveness) Mean aggressiveness ratings for the Regions with greater than 5 samples; groupings according to 95% family-wise confidence interval.

```{r MCG-ag, echo = FALSE, warning = FALSE, message = FALSE, results = "asis"}
read_csv("../../results/tables/MCG-aggressiveness.csv", 
         na = "NA", # spaces read in as spaces
         col_types = cols(.default = col_character())) %>%
  huxtable::as_huxtable(add_colnames = TRUE) %>%
  huxtable::set_number_format(row = huxtable::everywhere, col = 1, 0) %>% 
  huxtable::set_col_width(c(0.13, 0.25, 0.08)) %>%
  huxtable::set_align(row = huxtable::everywhere, col = 2, "right") %>%
  huxtable::print_md(max_width = 40)
```

Table: (\#tab:MCG-aggressiveness) Mean aggressiveness ratings for the first 10 MCG; groupings according to 95% family-wise confidence interval.

```{r MLG-ag, echo = FALSE, warning = FALSE, message = FALSE, results = "asis"}
read_csv("../../results/tables/MLG-aggressiveness.csv", 
         na = "NA", # spaces read in as spaces
         col_types = cols(.default = col_character())) %>%
  rename(MLH = MLG) %>%
  huxtable::as_huxtable(add_colnames = TRUE) %>%
  huxtable::set_number_format(row = huxtable::everywhere, col = 1, 0) %>% 
  huxtable::set_col_width(c(0.13, 0.25, 0.08)) %>%
  huxtable::set_align(row = huxtable::everywhere, col = 2, "right") %>%
  huxtable::print_md(max_width = 40)
```

Table: (\#tab:MLG-aggressiveness) Mean aggressiveness ratings for the first 10 MLH; groupings according to 95% family-wise confidence interval.

![
Graph showing complex associations between Mycelial Compatibility Groups (MCG) (dotted nodes) and Multilocus Haplotypes (MLH) (full nodes) where the number in each node represents the MLH/MCG assignment. 
Node size reflect the number of samples represented by each node (circle). 
Edges (arrows) point from MLH to MCG where the weight (thickness) of the edge represents the number of samples shared. 
Node color represents the community assignment based on the walktrap algorithm with a maximum of four steps [@pons2006computing]. 
](../../results/figures/publication/full-graph.pdf){#fullgraph width=100%}



<!-- community graph figure -->

![
Network of populations (nodes/circles) and their shared multilocus haplotypes (MLH) (edges/lines) haplotyped over 16 loci. 
Each node is labeled with **name (number of MLHs shared/number of MLHs total).** 
The shade and area of the nodes are proportional to the number of unique MLHs within the node and the inner nodes are proportional to the number of private MLHs to the region (bottom legend). 
Each edge represents a single MLH where its thickness represents the number of populations that share the MLH and the shade represents the value of $P_{sex}$, or the probability of encountering that MLH from two independent meiotic events.
](../../results/figures/publication/mlg-16.pdf){#community-graph16 width=100%} 

<!-- end fig -->


<!-- DAPC figure 3: The figure strikes back-->

![
Scatter plot of Discriminant Analysis of Principal Components on Regions and Years showing temporal variation across all Regions. 
Points (text labels) represent observed individuals connected to the population centroids with ellipses representing a 66% confidence interval for a normal distribution. 
The center of each component is represented as black grid lines.
](../../results/figures/publication/dapc_region_year.pdf){#DAPC-RY-FULL width=100%}

<!-- end figure -->

\newpage

# References {-}


